include/nbl/system/IFileArchive.h

```diff
@@ -6,19 +6,46 @@
 #ifndef __NBL_I_FILE_ARCHIVE_H_INCLUDED__
 #define __NBL_I_FILE_ARCHIVE_H_INCLUDED__
 
+
 #include "nbl/system/IFile.h"
 #include "nbl/system/CFileView.h"
-
+#include "nbl/system/IFileViewAllocator.h"
+#include "nbl/system/CFileViewVirtualAllocatorWin32.h"
 namespace nbl
 {
 
 namespace system
 {
 
+template<typename T>
+class CInnerArchiveFile : public CFileView<T>
+{
+	std::atomic_flag* alive;
+public:
+	CInnerArchiveFile(CFileView<T>* arch, std::atomic_flag* _flag) : CFileView<T>(std::move(*arch)), alive(_flag)
+	{
+		alive->test_and_set();
+		alive->notify_one();
```

Comment:
no need to notify anyone, just set it (notify it to wake up, nobody waits for the flag to turn TRUE, everyone waits for it to turn FALSE)

Reply:
also probably best idea to set the flag in an overloaded placement `new` operator